internal/core/metadata/rest_deviceprofile_test.go

```diff
@@ -123,6 +124,102 @@ func TestGetAllProfiles(t *testing.T) {
 	}
 }
 
+func TestAddDeviceProfile(t *testing.T) {
+	// this one uses validated because it needs it for input and the dbMock
+	emptyName := TestDeviceProfileValidated
+	emptyName.Name = ""
+
+	tests := []struct {
+		name           string
+		request        *http.Request
+		dbMock         interfaces.DBClient
+		vdcMock        MockValueDescriptorClient
+		expectedStatus int
+	}{
+		{
+			"OK",
+			createRequestWithBody(TestDeviceProfile),
```

Comment:
Drilling into this function it always returns a "PUT" request. This seems odd given that `restAddDeviceProfile` is mapped to a POST in router.go. Would prefer these to align. Solution is to specify `http.Method` as this function does `createRequestWithPathParameters(http.MethodGet, map[string]string{})`